apps/web/src/serlo-editor/plugins/fill-in-the-blanks-exercise/index.tsx

```diff
@@ -12,31 +11,40 @@ import { EditorPluginType } from '@/serlo-editor/types/editor-plugin-type'
 
 export type FillInTheBlanksMode = 'typing' | 'drag-and-drop'
 
+export type BlankId = string
+
+export type DraggableId = string
+
 export const fillInTheBlanksExercise: EditorPlugin<FillInTheBlanksExerciseState> =
   {
     Component: FillInTheBlanksExerciseEditor,
     config: {},
-    state: createFillInTheBlanksExerciseState(),
+    state: createState(),
   }
 
-export type FillInTheBlanksExerciseState = ReturnType<
-  typeof createFillInTheBlanksExerciseState
->
+export type FillInTheBlanksExerciseState = ReturnType<typeof createState>
 
-function createFillInTheBlanksExerciseState() {
+function createState() {
   const defaultMode: FillInTheBlanksMode = 'typing'
 
   return object({
     text: child({
       plugin: EditorPluginType.Text,
       config: {
         formattingOptions: [
-          ...defaultFormattingOptions,
+          TextEditorFormattingOption.code,
+          TextEditorFormattingOption.colors,
+          TextEditorFormattingOption.lists,
+          TextEditorFormattingOption.richTextBold,
+          TextEditorFormattingOption.richTextItalic,
           TextEditorFormattingOption.blank,
         ],
+        isInlineChildEditor: true,
       },
     }),
-    mode: string(defaultMode),
+    mode: string(defaultMode), // Currently there is no method for the user to switch to 'drag-and-drop'
+    // This could be used to store additional answers for mode drag-and-drop in the future
+    // additionalDraggableAnswers: optional(list(object({ answer: string() }))),
```

Comment:
Good question! I though about this as well. I would say no. We might have a drag-the-words exercise where the author wants to give specific hints for incorrect answers but does not want them to show up as draggables. Like a text that asks for all variants of the word 'go' (go, goes, going, ...) where 'gos' should have some specific feedback but should not show up as draggable.

Reply:
Thought about this but I am unsure! It totally makes sense that incorrect answers also cause draggables to appear for the user to drag. Otherwise, there would be no way (at least in drag-and-drop mode) for the user to answer with those. What I am unsure about is: What if the author adds an incorrect answer that is correct in another blank? Should there be two draggables then or just one?
I think we need to get input about that from UX and author perspective. I will ask them about it.